lib/Translation/EmitHLSCpp.cpp

```diff
@@ -1880,13 +1924,26 @@ void ModuleEmitter::emitModule(ModuleOp module) {
 #include <math.h>
 #include <stdint.h>
 #include <string.h>
+)XXX";
 
-using namespace std;
+  // Check for used Ip, emit corresponding include paths.
+  llvm::SmallDenseSet<Attribute> emittedIncludeAttrs;
+  module->walk([&](hls::InstanceOp instance) {
+    auto declaredOp = instance.getDeclareOp();
+    declaredOp.walk([&](hls::IncludeOp include) {
+      emittedIncludeAttrs.insert(include->getAttr("paths"));
+    });
+  });
+  for (const Attribute &curPath : emittedIncludeAttrs) {
+    os << "#include ";
+    os << curPath.dyn_cast<ArrayAttr>()[0];
+    os << "\n";
```

Comment:
One last thing, the attribute you should insert is actually an element of the `paths` array attribute. So here you should do something like:
```
llvm::SmallDenseSet<StringAttr> emittedIncludeAttr;
...
for (auto path : include.getPaths())
  emittedIncludeAttrs.insert(path.cast<StringAttr>());
```
And eventually you should emit it as something like `curPath.getValue().str()`.